indigo_app/views/works.py

```diff
@@ -56,6 +56,49 @@ def determine_place(self):
     def get_context_data(self, **kwargs):
         context = super(WorkViewBase, self).get_context_data(work=self.work, **kwargs)
         context['work_json'] = json.dumps(WorkSerializer(instance=self.work, context={'request': self.request}).data)
+
+        # add other dates to timeline
+        work_timeline = self.work.points_in_time()
+        other_dates = [
+            ('assent_date', self.work.assent_date),
+            ('commencement_date', self.work.commencement_date),
+            ('repealed_date', self.work.repealed_date)
+        ]
+        # add to existing dates (e.g. if publication and commencement dates are the same)
+        for entry in work_timeline:
+            for name, date in other_dates:
+                if entry['date'] == date:
+                    if name == 'commencement_date' and self.work.commencing_work:
+                        entry['commencement_date'] = True
+                        entry['commencing_work'] = self.work.commencing_work
+                    elif name == 'repealed_date' and self.work.repealed_by:
+                        entry['repealed_date'] = True
+                        entry['repealed_by'] = self.work.repealed_by
+                    else:
+                        entry[name] = True
+        # add new dates (e.g. if assent is before any of the other dates)
+        existing_dates = [entry['date'] for entry in work_timeline]
+        for name, date in other_dates:
+            if date and date not in existing_dates:
+                if name == 'commencement_date' and self.work.commencing_work:
```

Comment:
Similar comment from above applies here (but it's stylistic). You could start by doing everything in the final `else`, and then handle the exceptions.

Reply:
this way I don't have to then find the index of the dict in the list to update, so I was being a little lazy :grimacing: